### DISCUSSION TITLE
Missing glyphs when trying to get ligatures to work

### DISCUSSION THREAD

**Discussion by viandoxdev** [mooman219/fontdue#136] [Q&A]
I'm not sure this is the right place to ask, especially since I don't know if this is a fontdue or rustybuzz problem (It might not even be one and I'm just being stupid), but I couldn't find anywhere else.

I'm trying to do text rendering using fontdue and rustybuzz, and I'd like to support ligatures. but the glyphs index rustybuzz gives me, while valid (< `Font::glyph_count`), show up as zero sized (`metrics.width` and `metrics.height` are both zero). I'm not sure what could even be causing this problem, but unless rustybuzz is giving out some completely random glyphs, this looks to be coming from fontdue. I tried with two different font (Iosevka and FiraCode), in both cases, all normal glyphs work as intended, but all the ligatures (in my case the string "<=>") just don't show up.

Looks like this might be coming from [here](https://github.com/mooman219/fontdue/blob/ee59ba18f029596ca2315c09e509cb4da7aab097/src/font.rs#L296-L298), this only calls `generate_glyph` for glyphs that are mapped to a codepoint, which isn't the case for ligatures.

**Comment by mooman219** [comment]
Fixed with https://github.com/mooman219/fontdue/commit/0c3edfe3dfb9b06b65b04a1330e42bc7a84d6fa6
